internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -339,9 +339,14 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context, clust
 	if userManagedNetwork := isUserManagedNetwork(clusterDeployment); userManagedNetwork != swag.BoolValue(cluster.UserManagedNetworking) {
 		params.UserManagedNetworking = swag.Bool(userManagedNetwork)
 	}
-
-	// TODO: handle InstallConfigOverrides
-
+	// handle InstallConfigOverrides
+	updateClusterInstallConfigOverrides := false
```

Comment:
Because it's a different API i think that it should be done in a different function. 
It will be clearer. 
The `TODO` was here by mistake.

Reply:
Do you mean extracting the logic to a different function and call it from here? 

If not, why does it matter what is the underlying assisted-service API?
Both APIs update the cluster and this function is called `updateIfNeeded`

Reply:
Both options are fine. The part that i don't like is that we call different apis in the same function.